src/schemas/Polymorphic.js

```diff
@@ -45,6 +45,11 @@ export default class PolymorphicSchema {
       return value;
     }
     const schema = this.isSingleSchema ? this.schema : this.schema[value.schema];
-    return unvisit(value.id || value, schema, entities);
+    let idOrValue;
+    if (typeof schema.getId === 'function') {
```

Comment:
In a polymorphic transform, the `value` will always either be of the form: `{ id, schema }` or just the `id` value. 

Where are you coming across a different case? Can you include a test that proves this is needed?